src/Graphs/DirectedAcyclicGraphTransitiveReducer.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bearded.Utilities.Graphs
+{
+    public static class DirectedAcyclicGraphTransitiveReducer<T> where T : IEquatable<T>
+    {
+        public static IDirectedAcyclicGraph<T> ReduceGraph(IDirectedAcyclicGraph<T> graph)
+        {
+            var elements = graph.Elements as IList<T> ?? graph.Elements.ToList();
+            var graphBuilder = DirectedGraphBuilder<T>.NewBuilder();
+
+            foreach (var element in elements)
+            {
+                graphBuilder.AddElement(element);
+            }
+
+            foreach (var element in elements)
+            {
+                foreach (var neighbor in getOnlyDirectChildren(graph, element))
+                {
+                    graphBuilder.AddArrow(element, neighbor);
+                }
+            }
+
+            return graphBuilder.CreateAcyclicGraphUnsafe();
+        }
+
+        private static IEnumerable<T> getOnlyDirectChildren(IDirectedGraph<T> graph, T element)
```

Comment:
Wouldn't `getDirectOnlyChildren` be a better name? Or maybe `getDescendantsThatCanOnlyBeReachedDirectly` or something?

Reply:
The latter feels too long. This method "only gets the direct children". `getOnlyDirectChildren` sounds like the best translation of that in a method name. I wouldn't know what "direct only" would mean

Reply:
Maybe it is clear enough if you know what the algorithm is trying to do.

Reply:
Changed